In [7]:
# imports

import pandas as pd
import numpy as np
import statsmodels.api as sm


Build a regression model.

In [12]:
yelp_df = pd.read_csv('C:/Users/sherr/LHL classwork/LHL-P2/data/yelp_df.csv')
yelp_df = yelp_df.dropna()

# free bikes vs avg restaurant and pilates distances 

X = yelp_df[['avg_rdist', 'avg_pdist']]
y = yelp_df['free_bikes']

X = sm.add_constant(X)
lin_reg = sm.OLS(y, X)

Provide model output and an interpretation of the results. 

In [14]:
model = lin_reg.fit()
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:             free_bikes   R-squared:                       0.052
Model:                            OLS   Adj. R-squared:                  0.044
Method:                 Least Squares   F-statistic:                     6.900
Date:                Wed, 30 Apr 2025   Prob (F-statistic):            0.00121
Time:                        00:47:34   Log-Likelihood:                -777.53
No. Observations:                 257   AIC:                             1561.
Df Residuals:                     254   BIC:                             1572.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         11.8143      1.187      9.956      0.0

I am interested in examining the relationship between the number of free bikes a station has and averaege distance of the closest restaurant and pilates studio. A possible correlation - such as less bikes available when the closest restaurant or pilates studio is near, could mean that people are using bike shares to traverse to those locations. The regression model is a multivariate regression model that takes the average restaurant distance and average pilates studio distance as independent variables, and number of free bikes as the dependent variable. Since there are two independent variables, it's important to look at the adjusted R-squared instead of the regular R-square to account for the increase with each independent variable addition. An adjusted R-squared of 0.044 means that my model explains 4.4% of the variations in the data. The p-value for avg_rdist is 0.672, which is much greater than the usual 0.05 threshold, meaning the relationship between free bikes and avg_rdist is insignificant. The coefficient of both independent variables are similar but slightly higher for avg_pdist, meaning that for each unit increase in free bikes (e.g for each additional bike that is free), the average pilates studio is 0.0037 units closer. 

# Stretch

How can you turn the regression model into a classification model?